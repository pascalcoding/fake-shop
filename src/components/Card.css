.card {
  border: 3px solid #c4c4c4;
  border-radius: 10px;
  max-width: 300px;
  padding: 20px;
  text-decoration: none;
  color: #000;
}
