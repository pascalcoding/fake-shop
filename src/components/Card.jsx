import PropTypes from 'prop-types';
import { useState } from 'react';
import './Card.css';
import { Link } from 'react-router-dom';

function Card({ product }) {
  const [amount, setAmount] = useState(0);

  const handleClick = () => {};

  return (
    <Link className="card" to={`/Shop/product/${product.id}`}>
      <h3>{product.title}</h3>
      <img
        src={product.image}
        style={{ maxWidth: '150px', maxHeight: '100px' }}
      />
      <div>{product.price} â‚¬</div>
    </Link>
  );
}

Card.propTypes = {
  product: PropTypes.object,
};

export default Card;
