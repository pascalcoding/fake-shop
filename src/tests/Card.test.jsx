import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import Card from '../components/Card';
import { MemoryRouter } from 'react-router-dom';

describe('Card component', () => {
  it('Renders card correctly', () => {
    const product = {
      id: 1,
      title: 'Test Product 1',
      price: 100,
      category: 'Electronics',
      description: 'Test product 1 is very good',
      image: '',
    };
    const { container } = render(
      <MemoryRouter>
        <Card product={product} />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
