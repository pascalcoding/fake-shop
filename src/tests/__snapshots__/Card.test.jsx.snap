// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card component > Renders card correctly 1`] = `
<div>
  <a
    class="card"
    href="/Shop/product/1"
  >
    <h3>
      Test Product 1
    </h3>
    <img
      src=""
      style="max-width: 150px; max-height: 100px;"
    />
    <div>
      100
       â‚¬
    </div>
  </a>
</div>
`;
